Tests/LibWeb/Text/input/wpt-import/close-watcher/resources/helpers.js

```diff
@@ -0,0 +1,80 @@
+window.createRecordingCloseWatcher = (t, events, name, type, parentWatcher) => {
+  let watcher = null;
+  if (type === 'dialog') {
+    watcher = document.createElement('dialog');
+    watcher.textContent = 'hello world';
+    t.add_cleanup(() => watcher.remove());
+    if (parentWatcher?.appendChild) {
+      parentWatcher.appendChild(watcher);
+    } else {
+      document.body.appendChild(watcher);
+    }
+    watcher.showModal();
+  } else if (type === 'popover') {
+    watcher = document.createElement('div');
+    watcher.setAttribute('popover', 'auto');
+    watcher.textContent = 'hello world';
+    t.add_cleanup(() => watcher.remove());
+    if (parentWatcher?.appendChild) {
+      parentWatcher.appendChild(watcher);
+    } else {
+      document.body.appendChild(watcher);
+    }
+    watcher.showPopover();
+  } else {
+    watcher = new CloseWatcher();
+    t.add_cleanup(() => watcher.destroy());
+  }
+
+  const prefix = name === undefined ? '' : name + ' ';
+  watcher.addEventListener('cancel', e => {
+    const cancelable = e.cancelable ? '[cancelable=true]' : '[cancelable=false]';
+    events.push(prefix + 'cancel' + cancelable);
+  });
+  watcher.addEventListener('close', () => {
+    events.push(prefix + 'close');
+  });
+
+  return watcher;
+};
+
+window.createBlessedRecordingCloseWatcher = async (t, events, name, type, parentWatcher) => {
+  await maybeTopLayerBless(parentWatcher);
+  return createRecordingCloseWatcher(t, events, name, type, parentWatcher);
+};
+
+window.destroyCloseWatcher = (watcher) => {
+  if (watcher instanceof HTMLElement) {
+    watcher.remove();
+  } else {
+    watcher.destroy();
+  }
+};
+
+window.sendEscKey = () => {
+  // Esc is \uE00C, *not* \uu001B; see https://w3c.github.io/webdriver/#keyboard-actions.
+  //
+  // It's important to target document.body, and not any element that might stop receiving events
+  // if a popover or dialog is making that element inert.
+  return test_driver.send_keys(document.body, '\uE00C');
```

Comment:
Trying to run this test locally results in a timeout for me. 

This line is calling the WebDriver `send_keys` endpoint. Importing WPT tests that rely on WebDriver functionality isn't currently supported. 

FYI, we usually import tests using `./Meta/WPT.sh import http://wpt.live/close-watcher/abortsignal.html`, for example. If doing that doesn't work, then it's a good indication that running the test with `./Meta/ladybird.sh test LibWeb` will also fail.

As an alternative to importing this test, you could create a text test, which covers the same functionality as this test but doesn't require any user interaction.